Fix: session may not have been saved properly when redirected

zypA13510 opened the pull request:
jaredhanson/connect-ensure-login#28

Sorry it took a year to create this PR:smiley:.

zypA13510 committed:
fix: save session before redirect

Save the session to store before the redirect. Otherwise, the session
store may not save it in time.

Ref: mweibel/connect-session-sequelize#54

zypA13510 commented:
@rwky If you want, I think I can help maintain this.

zypA13510 committed:
ci: only runs on maintained node versions

zypA13510 committed:
test: rewrite all tests in mocha

zypA13510 committed:
chore: migrate to passport-next org

zypA13510 commented on README.md:
```diff
@@ -60,16 +60,21 @@ what happens:
-[![Build Status](https://secure.travis-ci.org/jaredhanson/connect-ensure-login.png)](http://travis-ci.org/jaredhanson/connect-ensure-login)
+[![Build Status](https://secure.travis-ci.org/passport-next/connect-ensure-login.svg?branch=master)](http://travis-ci.org/passport-next/connect-ensure-login)
+[![Coverage Status](https://coveralls.io/repos/github/passport-next/connect-ensure-login/badge.svg?branch=master)](https://coveralls.io/github/passport-next/connect-ensure-login?branch=master)
+<!--[![Maintainability](https://api.codeclimate.com/v1/badges/<token>/maintainability)](https://codeclimate.com/github/passport-next/connect-ensure-login/maintainability)-->
```
Token missing. Need to add this project to code climate first.

rwky commented:
Looks good, can you split this into to PRs one that fixes the issue and one that does the chore stuff. Makes it easier to review. I'll add the code climate integration

rwky commented:
Here's the code climate markdown `[![Maintainability](https://api.codeclimate.com/v1/badges/cd841c0f5037adb4d874/maintainability)](https://codeclimate.com/github/passport-next/connect-ensure-login/maintainability)`

zypA13510 commented:
> can you split this into to PRs one that fixes the issue and one that does the chore stuff. Makes it easier to review.

Actually I split them by commits. You can view individual commits and it should be more obvious what each commit is doing.

Btw, I put them all in one PR because the old test framework always fails, and I don't know how to fix it. So I have rewritten the tests. If they are in a separate PR, this PR will fail the tests. But if you prefer, I can split it into a separate PR.

zypA13510 committed:
docs: update maintainability badge

rwky commented:
ok fair enough, to save me time rummaging through the commits, did you add a test for the fix you added?

zypA13510 commented:
> did you add a test for the fix you added

Fair point, I did not. Will add it next week.

Maybe I will do it like this: first I will close this PR, and create a new one that only contains rebranding and test-fixing stuff. And then another with this fix and the related tests. Is this ok to you?

rwky commented:
Sounds good to me!

On Sun, 8 Sep 2019, at 09:57, Yuping Zuo wrote:
> >  
> > did you add a test for the fix you added
> >  Fair point, I did not. Will add it next week.
> 
> Maybe I will do it like this: first I will close this PR, and create a 
> new one that only contains rebranding and test-fixing stuff. And then 
> another with this fix and the related tests. Is this ok to you?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/passport-next/connect-ensure-login/pull/1?email_source=notifications&email_token=AABFBI2YPGLKKS7DZWINJZLQIS477A5CNFSM4IKGPXHKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6FLIHI#issuecomment-529183773>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AABFBI22Z7AZGREX5DVPGCLQIS477ANCNFSM4IKGPXHA>.
>

-- 
Regards

Rowan Wookey BSc (Hons) Comp, CISMP, MBCS 
Server Administrator & Programmer

Please add admin@rwky.net to your contacts/email whitelist.
